providers/dns/nearlyfreespeech/nearlyfreespeech.go

```diff
@@ -144,56 +132,27 @@ func (d *DNSProvider) Present(domain, token, keyAuth string) error {
 
 // CleanUp removes the TXT record matching the specified parameters.
 func (d *DNSProvider) CleanUp(domain, token, keyAuth string) error {
-	ch, err := newChallenge(domain, keyAuth)
+	fqdn, value := dns01.GetRecord(domain, keyAuth)
+
+	authZone, err := dns01.FindZoneByFqdn(dns01.ToFqdn(domain))
```

Comment:
The `dns01.ToFqdn` function is causing a single period `.` character to appear at the end of the domain name string passed into client.go to build API request URLs (ie `/dns/nfsdns.kittykatt.co/addRR`).

Using as is results in paths like:
`/dns/nfsdns.kittykatt.co./addRR`

Using `domain` to pass into `d.client.AddRecord` and `d.client.RemoveRecord` results in the correct paths:
`/dns/nfsdns.kittykatt.co/addRR`

I'm unsure what function it serves to mutate the domain name string to include that last dot in this specific case, but it definitely is resulting in bad request URIs